packages/tupaia-web/src/features/EnlargedDashboardItem/utils/useExportDashboardItem.tsx

```diff
@@ -0,0 +1,185 @@
+  entityName?: Entity['name'],
+  exportRef?: RefObject<HTMLElement>,
+) => {
+  const { projectCode, entityCode } = useParams();
+  const {
+    exportFormat,
+    exportWithLabels,
+    exportWithTable,
+    exportWithTableDisabled,
+    isExporting,
+    exportError,
+    isExportMode,
+  } = useContext(ExportContext);
+  const dispatch = useContext(ExportDispatchContext)!;
+  const { config } = currentDashboardItem || ({} as DashboardItem);
+  const { type, presentationOptions, name } = config || ({} as DashboardItemConfig);
+  const exportTitle = `${name}, ${entityName}`;
+
+  const { doExport } = useChartDataExport(
+    {
+      ...report,
+      ...config,
+      presentationOptions: {
+        ...(presentationOptions || {}),
+        exportWithLabels,
+        exportWithTable,
+        exportWithTableDisabled,
+      },
+    } as ViewContent,
+    exportTitle,
+  );
+
+  // use the useExportToExcel hook to export the matrix dashboard item. We disable the fetch initially so that we can only do the query when we want to export
+  const { error: excelError, refetch: callExportToExcel, isFetching } = useExportToExcel(
+    getExportToExcelParams({
+      projectCode,
+      dashboardCode: activeDashboard?.code,
+      entityCode,
+      config,
+      report,
+      currentDashboardItem,
+    }),
+    false,
+  );
+
+  const filename = toFilename(`export-${entityName}-${name}`, true);
+  const file = `${filename}.${exportFormat}`;
+
+  const setExportError = (message: string | null) => {
+    dispatch({ type: ACTION_TYPES.SET_EXPORT_ERROR, payload: message });
+  };
+
+  const setIsExporting = (isExporting: boolean) => {
+    dispatch({ type: ACTION_TYPES.SET_IS_EXPORTING, payload: isExporting });
+  };
+
+  // Even though we could get the data from the useExportToExcel hook directly, we will use the result of the callExportToExcel function to get the data, because we need to only download the data when the user clicks the export button. Putting it inside a useEffect hook makes it hard to control exactly when the download happens
+  const exportMatrixToExcel = async () => {
+    const result = await callExportToExcel();
+    downloadJs(result.data, file);
+  };
+
+  // export to png if the export format is png. This should handle the setting of the isExporting state and the export error state because it is not using a hook from elsewhere
+  const exportToPNG = async () => {
+    try {
+      setIsExporting(true);
+      const node = exportRef?.current as Node;
+      const result = await domtoimage.toPng(node, { bgcolor: 'white' });
+      downloadJs(result, file);
+    } catch (e: any) {
+      setExportError(e.message);
+    } finally {
+      setIsExporting(false);
+    }
+  };
+
+  const EXPORT_FUNCTIONS = {
+    [EXPORT_FORMATS.PNG]: exportToPNG,
+    [EXPORT_FORMATS.XLSX]: type === 'matrix' ? exportMatrixToExcel : doExport,
+  };
+
+  // set export error message if an error is caught from the useExportToExcel hook
+  useEffect(() => {
```

Comment:
Once the useExport hook is refactored to a mutation there should be a nicer way to do these side effects such as the onError handler or you can use a try catch block if that suits better.